test/core/xds/xds_audit_logger_registry_test.cc

```diff
@@ -0,0 +1,119 @@
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+//
+
+#include "src/core/ext/xds/xds_audit_logger_registry.h"
+
+#include <string>
+
+#include <google/protobuf/any.pb.h>
+
+#include "absl/status/status.h"
+#include "absl/status/statusor.h"
+#include "envoy/config/rbac/v3/rbac.upb.h"
+#include "gtest/gtest.h"
+#include "upb/def.hpp"  // IWYU pragma: keep
+#include "upb/upb.hpp"
+
+#include <grpc/grpc.h>
+
+#include "src/core/ext/xds/xds_bootstrap_grpc.h"
+#include "src/core/lib/json/json_writer.h"
+#include "src/proto/grpc/testing/xds/v3/extension.pb.h"
+#include "src/proto/grpc/testing/xds/v3/rbac.pb.h"
+#include "src/proto/grpc/testing/xds/v3/stream.pb.h"
+#include "test/core/util/test_config.h"
+
+// IWYU pragma: no_include "upb/reflection/def.hpp"
+
+namespace grpc_core {
+namespace testing {
+namespace {
+
+using AuditLoggerConfigProto =
+    ::envoy::config::rbac::v3::RBAC::AuditLoggingOptions::AuditLoggerConfig;
+using ::envoy::extensions::rbac::audit_loggers::stream::v3::StdoutAuditLog;
+
+absl::StatusOr<std::string> ConvertAuditLoggerConfig(
+    const AuditLoggerConfigProto& config) {
+  std::string serialized_config = config.SerializeAsString();
+  upb::Arena arena;
+  upb::SymbolTable symtab;
+  XdsResourceType::DecodeContext context = {nullptr,
+                                            GrpcXdsBootstrap::GrpcXdsServer(),
+                                            nullptr, symtab.ptr(), arena.ptr()};
+  auto* upb_config =
+      envoy_config_rbac_v3_RBAC_AuditLoggingOptions_AuditLoggerConfig_parse(
+          serialized_config.data(), serialized_config.size(), arena.ptr());
+  ValidationErrors errors;
+  ValidationErrors::ScopedField field(&errors, ".logger_config");
+  auto config_json = XdsAuditLoggerRegistry().ConvertXdsAuditLoggerConfig(
+      context, upb_config, &errors);
+  if (!errors.ok()) {
+    return errors.status(absl::StatusCode::kInvalidArgument,
+                         "validation errors");
+  }
+  return JsonDump(config_json);
+}
+
+//
+// StdoutLoggerTest
+//
+
+TEST(StdoutLoggerTest, Basic) {
+  AuditLoggerConfigProto config;
+  config.mutable_audit_logger()->mutable_typed_config()->PackFrom(
+      StdoutAuditLog());
+  auto result = ConvertAuditLoggerConfig(config);
+  ASSERT_TRUE(result.ok()) << result.status();
+  EXPECT_EQ(*result, "{\"stdout_logger\":{}}");
+}
+
+//
+// XdsAuditLoggerRegistryTest
+//
+
+TEST(XdsAuditLoggerRegistryTest, EmptyAuditLoggerConfig) {
+  auto result = ConvertAuditLoggerConfig(AuditLoggerConfigProto());
+  EXPECT_EQ(result.status().code(), absl::StatusCode::kOk);
+  EXPECT_EQ(*result, "null");
+}
+
+TEST(XdsAuditLoggerRegistryTest, NoSupportedType) {
```

Comment:
Because this API takes one logger at a time, this test case you referred to is basically the `EmptyAuditLoggerConfig` above, I changed it to check for errors now.

I should have test cases that cover all the possible errors this function can return (for now).

Reply:
Added. Thanks!